### DISCUSSION TITLE
How to check all properties and corresponding values of UNISIM cell?

### DISCUSSION THREAD

**Discussion by dg-deus** [Xilinx/RapidWright#879] [Q&A] [answered]
I can create a cell and check its all properties and corresponding values with Xilinx TCL commands like the following codes as an example.
```
set Cell [create_cell -referece FDCE "myCell"]
set PropList [list_property $Cell]
foreach Prop $PropList {
  set PropVal [list_property_value $Prop $Cell]
}
```

I want to do something similar with RapidWright, because I need to know what possible properties and corresponding values exist for target cell first before setting cell's properties.

I tried something like the codes below in python. However, it seems like only default values are available.
How can I get all properties and corresponding values of UNISIM cells in RapidWright?
```
RwDevice = Device.getDevice(device)
RwDesign = Design("myDesign", RwDevice.getName())
RwCell = RwDesign.createCell("myCell", Unisim.FDCE)
for prop, def_val in RwDesign.getDefaultCellProperties(Series.Series7, RwCell.getType()).items():
  print(f"Prop: {prop}, Default Value: {def_val}, Current Value: {RwCell.getProperty(prop)}")
```

Thank you.

**Comment by clavin-xlnx** [comment]
Take a look at this comment for non-default properties: 
https://github.com/Xilinx/RapidWright/issues/67#issuecomment-667295332

**Reply by dg-deus** [reply]
Thanks, and I also had a look at the comment before.

Based on the comment, the properties of primitive cells are not present in RapidWrigth. So I cannot know what kind of values I can use for given properties without cross checking Vivado, right?

Can I look forward to RapidWrigt in the future also providing methods for this application?

Thank you.

**Reply by clavin-xlnx** [reply, marked as answer]
You are correct, we don't have exhaustive lists of properties settings in RapidWright.  This could be added in the future.  I'll look into the difficulty of adding them.  For, now, however, you'll probably just need to query in Vivado.
